src/it/streamingcommunity/src/eu/kanade/tachiyomi/animeextension/it/streamingcommunity/StreamingCommunity.kt

```diff
@@ -252,68 +256,80 @@ class StreamingCommunity : ConfigurableAnimeSource, AnimeHttpSource() {
+                ?: error("Failed to extract iframe")
+        val iframeHeaders =
+            headers
+                .newBuilder()
+                .add("Accept", "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8")
+                .add("Host", iframeUrl.toHttpUrl().host)
+                .add("Referer", "$baseUrl/")
+                .build()
+
+        val iframe =
+            client
+                .newCall(
+                    GET(iframeUrl, headers = iframeHeaders),
+                ).execute()
+                .asJsoup()
+        val script = iframe.selectFirst("script:containsData(masterPlaylist)")!!.data().replace("\n", "\t")
+        var playlistUrl = Regex("""url: ?'(.*?)'""").find(script)!!.groupValues[1]
+        val filename = playlistUrl.slice(playlistUrl.lastIndexOf("/") + 1 until playlistUrl.length)
+        if (!filename.endsWith(".m3u8")) {
+            playlistUrl = playlistUrl.replace(filename, filename + ".m3u8")
+        }
 
-        val embedUrl = data.props.embedUrl
-        val embedHeaders = headers.newBuilder()
-            .add("Accept", "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8")
-            .add("Host", baseUrl.toHttpUrl().host)
-            .add("Referer", response.request.url.toString())
-            .build()
-
-        val embedded = client.newCall(
-            GET(embedUrl, headers = embedHeaders),
-        ).execute().asJsoup()
-
-        val iframeUrl = embedded.selectFirst("iframe[src]")?.attr("abs:src") ?: error("Failed to load iframe")
-        val iframeHeaders = headers.newBuilder()
-            .add("Accept", "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8")
-            .add("Host", iframeUrl.toHttpUrl().host)
-            .add("Referer", "$baseUrl/")
-            .build()
-
-        val iframe = client.newCall(
-            GET(iframeUrl, headers = iframeHeaders),
-        ).execute().asJsoup()
-        val script = iframe.selectFirst("script:containsData(masterPlaylistParams)")!!.data()
-
-        val playlistUrl = Regex("""masterPlaylistUrl.*?'(.*?)'""").find(script)!!.groupValues[1]
         val expires = Regex("""'expires': ?'(\d+)'""").find(script)!!.groupValues[1]
-        val canCast = Regex("""'canCast': ?'(\d*)'""").find(script)!!.groupValues[1]
         val token = Regex("""'token': ?'([\w-]+)'""").find(script)!!.groupValues[1]
 
         // Get subtitles
-        val masterPlUrl = "$playlistUrl?token=$token&expires=$expires&canCast=$canCast&n=1"
-        val masterPl = client.newCall(GET(masterPlUrl)).execute().body.string()
-        val subList = Regex("""#EXT-X-MEDIA:TYPE=SUBTITLES.*?NAME="(.*?)".*?URI="(.*?)"""").findAll(masterPl).map {
-            Track(it.groupValues[2], it.groupValues[1])
-        }.toList()
-
+        val masterPlUrl = "$playlistUrl?token=$token&expires=$expires&n=1"
+        val masterPl =
```

Comment:
Maybe use the playlistutils lib?

Reply:
I had to do it manually because I got errors xD